crates/bevy_render/src/maths.wgsl

```diff
@@ -27,3 +27,16 @@ fn mat2x4_f32_to_mat3x3_unpack(
         vec3<f32>(a[1].zw, b),
     );
 }
+
+// Creates an orthonormal basis given a Z vector and an up vector (which becomes
+// Y after orthonormalization).
+//
+// The results are equivalent to the Gram-Schmidt process [1].
+//
+// [1]: https://math.stackexchange.com/a/1849294
+fn orthonormalize(z_unnormalized: vec3<f32>, up: vec3<f32>) -> mat3x3<f32> {
+    let z_basis = normalize(z_unnormalized);
+    let x_basis = normalize(cross(z_basis, up));
```

Comment:
I’m trying to think of this necessarily works for both left- and right-handed coordinate systems. I think it does…

Reply:
It will, in that it will produce *some* orthonormal basis. It might not be the orthonormal basis that you want it to, though, if you have some extra requirements. In our case it works because we don't really care about what the orthonormal basis is as long as the Z vector is retained.